third_party/terraform/resources/resource_composer_environment.go.erb

```diff
@@ -153,6 +153,46 @@ func resourceComposerEnvironment() *schema.Resource {
 										},
 										Set: schema.HashString,
 									},
+									"ip_allocation_policy": {
+										Type: schema.TypeList,
+										Optional: true,
+										Computed: true,
+										ForceNew: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"use_ip_aliases": {
+													Type:     schema.TypeBool,
+													Optional: true,
+													ForceNew: true,
+												},
+												"cluster_secondary_range_name": {
+													Type:     schema.TypeString,
+													Optional: true,
+													ForceNew: true,
+													ConflictsWith: []string{"config.0.node_config.0.ip_allocation_policy.0.cluster_ipv4_cidr_block"},
+												},
+												"services_secondary_range_name": {
+													Type:     schema.TypeString,
+													Optional: true,
+													ForceNew: true,
+													ConflictsWith: []string{"config.0.node_config.0.ip_allocation_policy.0.services_ipv4_cidr_block"},
+												},
+												"cluster_ipv4_cidr_block": {
+													Type:     schema.TypeString,
+													Optional: true,
+													ForceNew: true,
+													ConflictsWith: []string{"config.0.node_config.0.ip_allocation_policy.0.cluster_secondary_range_name"},
```

Comment:
We might also need to add the diff suppress cidrOrSizeDiffSuppress for the cidr block fields. Most of my comments are based off what cluster seems to include in ip_allocation_policy :)

Reply:
Ah I never considered comparing this to the base cluster resource. Done.